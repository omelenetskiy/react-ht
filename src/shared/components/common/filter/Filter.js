import React, { Component } from "react";
import { connect } from "react-redux";

class Filter extends Component {
  state = {};

  render() {
    const {
      movies: { movies }
    } = this.props;

    return (
      <section className="filter">
        <span className="navbar-text">{`${movies &&
          movies.length} movies found`}</span>
        <div>
          <span className="navbar-text">Sort by: </span>{" "}
          <div className="btn-group btn-group-toggle" data-toggle="buttons">
            <label className="btn btn-danger btn-sm active">
              <input
                type="radio"
                name="options"
                checked
                onChange={() => console.log("changed")}
              />{" "}
              release date
            </label>
            <label className="btn btn-danger btn-sm">
              <input
                type="radio"
                name="options"
                checked
                onChange={() => console.log("changed")}
              />{" "}
              raiting
            </label>
          </div>
        </div>
      </section>
    );
  }
}

function moviesState(state) {
  return {
    movies: state.movies
  };
}

export default connect(moviesState)(Filter);
