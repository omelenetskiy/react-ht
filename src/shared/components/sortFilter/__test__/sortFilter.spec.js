import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import 'jest-enzyme';
import 'jest-styled-components';

import Filter from '../SortFilter';

const initialState = {};
const mockStore = configureStore();
let store;

const changeSortBy = jest.fn();
const props = {
  changeSortBy,
  movies: ['movie'],
  match: {
    params: {
      query: 'query',
    },
  },
};

describe('Filter', () => {
  beforeEach(() => {
    store = mockStore(initialState);
  });
  it('should render correctly', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Router>
          <Filter {...props} changeSortBy={changeSortBy} />
        </Router>
      </Provider>
    );
    // wrapper.instance().toggleSortBy({ target: { value: 'cats' } });

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
