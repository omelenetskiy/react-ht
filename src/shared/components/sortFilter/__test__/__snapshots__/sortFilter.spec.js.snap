// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter should render correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeState": Object {},
    }
  }
>
  <BrowserRouter>
    <withRouter(Connect(Filter))
      changeSortBy={[MockFunction]}
      match={
        Object {
          "params": Object {
            "query": "query",
          },
        }
      }
      movies={
        Array [
          "movie",
        ]
      }
    />
  </BrowserRouter>
</ContextProvider>
`;
